### PR TITLE
Clarify Hub generated syncerror events

### PR DESCRIPTION
**Pull request by EricOnFHIR**
Based on discussion in February 2022 Hackathon, propose clarification on Hub generated syncerror events

### CODE REVIEW

File: input/pagecontent/2-5-EventNotification.md [hunk 1]
```diff
@@ -107,12 +107,34 @@ Field | Optionality | Type | Description
...
 
 {% include img.html img="ErrorSequence.png" caption="Figure: Event Notification Error flow diagram" %}
 
-More information on the source of notification errors and how to resolve them can be found in [Synchronization Considerations](4-2-syncconsiderations.html).
\ No newline at end of file
+More information on the source of notification errors and how to resolve them can be found in [Synchronization Considerations](4-2-syncconsiderations.html).
+
+## Hub Generated `syncerror` Events
+
+In addition to distributing [`syncerror`](3-8-syncerror.html) events sent by a subscribed application to all subscribed applications, the Hub may generate [`syncerror`](3-8-syncerror.html) events under certain conditions and distribute these events to all subscribed applications.
```
Comment (isaacvetter) [inline input/pagecontent/2-5-EventNotification.md:122]:
```suggestion
In addition to distributing [`syncerror`](3-8-syncerror.html) events sent by a subscribed application to all subscribed applications, the Hub SHALL generate and communicate [`syncerror`](3-8-syncerror.html) events to subscribed applications under the following conditions -- 
```
Clarify language for readability.

File: input/pagecontent/2-5-EventNotification.md [hunk 2]
```diff
@@ -107,12 +107,34 @@ Field | Optionality | Type | Description
...
+
+The Hub SHALL generate [`syncerror`](3-8-syncerror.html) events when a request requiring events be sent and a subscribed application:
```
Comment (isaacvetter) [inline input/pagecontent/2-5-EventNotification.md:124]:
```suggestion

```

File: input/pagecontent/2-5-EventNotification.md [hunk 3]
```diff
@@ -107,12 +107,34 @@ Field | Optionality | Type | Description
...
+1.  has previously closed its WebSocket connection to the Hub with a Close Reason other than 1000 or 1001 (see [WebSocket RFC](https://www.rfc-editor.org/rfc/rfc6455.html#section-7.1.6))
```
Comment (isaacvetter) [inline input/pagecontent/2-5-EventNotification.md:125]:
```suggestion
1.  A subscribed application's WebSocket connection is closed with any Connection Close Reason other than 1000 (normal closure) or 1001 (going away) (see [WebSocket RFC](https://www.rfc-editor.org/rfc/rfc6455.html#section-7.1.6) and [WebSocket Status Codes](https://www.rfc-editor.org/rfc/rfc6455.html#section-7.4))

```

File: input/pagecontent/2-5-EventNotification.md [hunk 4]
```diff
@@ -107,12 +107,34 @@ Field | Optionality | Type | Description
...
+2. does not respond to the distributed event by the Hub within 10 seconds
+
+Generated [`syncerror`](3-8-syncerror.html) events should be distributed only to applications which have subscribed to the event which triggered the [`syncerror`](3-8-syncerror.html) to be generated.
+
+After the Hub generates and distributes a [`syncerror`](3-8-syncerror.html) indicating that an application is not responsive, the Hub shall remove that application from its list of applications subscribed to the topic so that no additional [`syncerror`](3-8-syncerror.html) events are distributed on behalf of the nonresponsive application.
+
+The Hub SHALL NOT generate [`syncerror`](3-8-syncerror.html) events in the following situations:
+1. If a client fails to respond to a [`heartbeat`](3-11-heartbeat.html) event
+2. If a client closes its WebSocket connection to the Hub with a Close Reason of 1000 or 1001 (see [WebSocket RFC](https://www.rfc-editor.org/rfc/rfc6455.html#section-7.1.6)).  Note that if an application is deliberately shutdown, the application should unsubscribe (Unscrube event here) as part of their shutdown process (thus avoiding this scenario).
```
Comment (isaacvetter) [inline input/pagecontent/2-5-EventNotification.md:134]:
```suggestion
2. The application closes its WebSocket connection to the Hub with a [Connection Close Reason](https://www.rfc-editor.org/rfc/rfc6455.html#section-7.4.1) of 1000 (normal closure) or 1001 (going away).  

During a normal shutdown of an application, it SHALL unsubscribe, and provide a WebSocket Connection Close Reason of 1000 and not rely upon the Hub recognizing and unsubscribing it as an unresponsive app.

```

File: input/pagecontent/2-5-EventNotification.md [hunk 5]
```diff
@@ -107,12 +107,34 @@ Field | Optionality | Type | Description
...
\ No newline at end of file
+More information on the source of notification errors and how to resolve them can be found in [Synchronization Considerations](4-2-syncconsiderations.html).
+
+## Hub Generated `syncerror` Events
+
+In addition to distributing [`syncerror`](3-8-syncerror.html) events sent by one application to other subscribed applications, the Hub MAY generate and communicate [`syncerror`](3-8-syncerror.html) events to applications under the following conditions -- 
+
+1.  A subscribed application's WebSocket connection is closed with any Connection Close Reason other than 1000 (normal closure) or 1001 (going away) (see [WebSocket RFC](https://www.rfc-editor.org/rfc/rfc6455.html#section-7.1.6) and [WebSocket Status Codes](https://www.rfc-editor.org/rfc/rfc6455.html#section-7.4))
+
+2. does not respond to the distributed event by the Hub within 10 seconds
```
Comment (isaacvetter) [inline input/pagecontent/2-5-EventNotification.md]:
```suggestion
2. And, the application does not respond to the distributed event by the Hub within 10 seconds or an order of magnitude lower than the subscription time-out.

Implementer feedback is solicited on the amount and specificity of time, above.

```

**Comment by EricOnFHIR** [comment]
Close in favor of PR391

**Pull request closed without merging**